clients/nodejs/cac-client/client.ts

```diff
@@ -0,0 +1,106 @@
+    platform == "linux" ?
+    '../../../target/debug/libcac_client.dll' :
+    '../../../target/debug/libcac_client.so';
+
+const libPath = path.join(__dirname, file);
+
+const refType = ref.types;
+const int = refType.int;
+const string = refType.CString;
+const voidType = refType.void;
+
+class CacClient {
+    tenant: string | null = null;
+    cacHostName: string | null = null;
+    pollingFrequency: number = 10;
+
+    rustLib = ffi.Library(libPath, {
+        'cac_new_client': [int, [string, int, string]],
+        'cac_get_client': [string, [string]],
+        'cac_start_polling_update': [voidType, [string]],
+        'cac_free_client': [voidType, [string]],
+        'cac_last_error_message': [string, []],
+        'cac_get_config': [string, [string, string, string]],
+        'cac_last_error_length': [int, [voidType]],
+        'cac_free_string': [voidType, [string]],
+        'cac_get_last_modified': [string, [string]],
+        'cac_get_resolved_config': [string, [string, string, string, string]],
+        'cac_get_default_config': [string, [string, string]],
+
+    });
+
+    constructor(tenantName: string, pollingFrequency: number, cacHostName: string) {
+        if (tenantName == "" || cacHostName == "") {
+            throw Error("tenantName cannot be null or empty")
+        }
+        this.tenant = tenantName;
+        this.pollingFrequency = pollingFrequency;
+        this.cacHostName = cacHostName;
+    }
+
+    public getCACLastErrorMessage(): string | null {
+        return this.rustLib.cac_last_error_message();
+    }
+
+    public getCACLastErrorLength(): number {
+        return this.rustLib.cac_last_error_length()
+    }
+
+    public getCACClient(): string | null {
+        return this.rustLib.cac_get_client(this.tenant);
+    }
+
+    public createNewCACClient(): number {
+        let resp = this.rustLib.cac_new_client(this.tenant, this.pollingFrequency, this.cacHostName);
+        if (resp == 1) {
+            let errorMessage = this.getCACLastErrorMessage();
+            console.error("Some Error Occur while creating new client ", errorMessage);
+        }
+        return resp;
+    }
+
+    public async startCACPollingUpdate() {
+        this.rustLib.cac_start_polling_update(this.tenant)
+    }
+
+    public getCACConfig(filterQuery: string, filterPrefix: string): string | null {
+        let clientPtr = this.getCACClient();
+        console.log("cac_get_config", this.getCACLastErrorMessage());
+        return this.rustLib.cac_get_config(clientPtr, filterQuery, filterPrefix);
+    }
+
+    public freeCACClient(clientPtr: string) {
+        this.rustLib.cac_free_client(clientPtr)
+    }
+
+    public freeCACString(str: string) {
+        this.rustLib.cac_free_string(str)
+    }
+
+    public getLastModified(): string | null {
+        return this.rustLib.cac_get_last_modified(this.getCACClient());
+    }
+
+    public getResolvedConfig(query: string, filterKeys: string, mergeStrategy: string): string | null {
```

Comment:
Why return type for all functions is string, I am assuming its a JSON string?
Can we define types for these and de-serialize and send an actual object back?